packages/svelte/src/internal/client/dom/elements/misc.js

```diff
@@ -42,14 +42,19 @@ export function add_form_reset_listener() {
 			(evt) => {
 				// Needs to happen one tick later or else the dom properties of the form
 				// elements have not updated to their reset values yet
-				Promise.resolve().then(() => {
-					if (!evt.defaultPrevented) {
-						for (const e of /**@type {HTMLFormElement} */ (evt.target).elements) {
-							// @ts-expect-error
-							e.__on_r?.();
+				Promise.resolve()
+					.then(() => {
+						if (!evt.defaultPrevented) {
+							for (const e of /**@type {HTMLFormElement} */ (evt.target).elements) {
+								// @ts-expect-error
+								e.__on_r?.();
+							}
 						}
-					}
-				});
+					})
+					.catch((err) => {
```

Comment:
this block is unreachable, if Promise.resolve() endered catch, the engine would be broken and you'd have much worse problems everywhere

Reply:
Excellent point. I filed an issue for it here: https://github.com/typescript-eslint/typescript-eslint/issues/9192

Reply:
Actually, per the response there, we need to worry about the `.then()` failing